core/src/test/resources/sql/functions.iq

```diff
@@ -938,5 +938,20 @@ table(AUX.TBLFUN_IDENTITY(3)) as t3(v);
 
 !ok
 
+# SUBSTRING
+-- returns 'null'
+select SUBSTRING(NULL FROM 1 FOR 2);
+select SUBSTRING('text' FROM 1 FOR NULL);
+select SUBSTRING('text' FROM NULL FOR 2);
+select SUBSTRING(s from i for l) from (values ('abc', null, 2)) as t (s, i, l);
```

Comment:
minor comment, better to unify the usage of uppercase and lowercase. You could follow the test cases above.

Reply:
done